### PR TITLE
chore(web): relocations and cleanups

### PR DESCRIPTION
**Pull request by fabricionaweb**
We are having a warn message from Vite because having only one thick js file isnt ideal.

before

and it will continue complaing

### CODE REVIEW

File: web/vite.config.ts [hunk 1]
```diff
@@ -1,38 +1,30 @@
...
       useCredentials: true,
-      includeAssets: [
-        "favicon.svg",
-        "favicon.ico",
-        "robots.txt",
-        "logo.png",
-        "apple-touch-icon-*.png",
-        "manifest.webmanifest",
-        "assets/**/*"
-      ],
```
Comment (fabricionaweb) [inline web/vite.config.ts:28]: I see the docs, the `includeAssets` by default already gets public folder and icons... I tested with and without this code and the generated manifest had no changes.

File: web/vite.config.ts [hunk 2]
```diff
@@ -98,7 +90,25 @@ export default ({ mode }: { mode: any }) => {
     },
     build: {
       manifest: true,
-      sourcemap: true
+      sourcemap: true,
+      rollupOptions: {
+        output: {
+          manualChunks (id: string) {
```
Comment (fabricionaweb) [inline web/vite.config.ts:96]: This `manualChunks` is kinda weird, but without it, Vite by default only splits `index` and `vendor`, which isnt good enough to me.

File: web/vite.config.ts (hunk 1 above)
Comment (fabricionaweb) [reply web/vite.config.ts:28]: Actually let me do more tests

File: web/pnpm-lock.yaml [hunk 3]
```diff
@@ -159,7 +159,7 @@ packages:
     dependencies:
       '@jridgewell/gen-mapping': 0.3.3
       '@jridgewell/trace-mapping': 0.3.18
-    dev: false
+    dev: true
```
[the remark this answers is not in the document]
Comment (fabricionaweb) [reply web/pnpm-lock.yaml:162]: Because I moved the vite-plugin to devDependencies

File: web/vite.config.ts (hunk 2 above)
Comment (fabricionaweb) [reply web/vite.config.ts:104]: Yes, but to be honest, I could not understand well 🤓

And the `manualChunks` does not work without the plugin

File: web/pnpm-lock.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (fabricionaweb) [reply web/pnpm-lock.yaml:162]: To be honest we dont follow strategy about what is dev or not in this app, as the whole web will always need to build and the runtime is done without any node dependencies. Maybe we should move everything to devDependencies (or dependencies). 

That makes sense to me

[the remark this answers is not in the document]
Comment (fabricionaweb) [reply web/pnpm-lock.yaml:162]: I will drop this commit and we see/decide it in another MR

**Review by zze0s — COMMENTED** [review]
I think it's worth exploring using lazy imports in the router to only load per page. https://react.dev/reference/react/lazy

**Comment by stacksmash76** [comment]
Late to the party, but my 5 cents:
- React Suspense should be implemented first before lazy loading.
- Too much chunking is a bad practice (relative to project size, of course; you don't want a 1 MB file split across more than, say, 10 files). You've got ping roundabout (out & back) plus HTTP overhead response plus V8 doing it's thang for every little navigation [the browser can't burst the requests in some cases, see below]. Also, cache busting can be tedious/a nuisance.
- When going the React.lazy route, only the main chunks are embedded into the HTML (i.e. manualChunks in this instance), which is a bullet-proof way of loading JS. Problems occur when you load chunks from JS. There are quite a few browser bugs, with differing errors (source: ancedata from one of my Sentry instances where I experimented with this; sadly, deleted, can't give you concrete/concise examples). This happens on Chrome and Safari predominantly. For Firefox this is a rarer occurrence (when normalized to sampled distribution).

Most of the time you're serving gzip (or in case of newer browsers brotli anyways). So, improve terser/optimizer to conform to the GUWOP web standard:
- 3 compression passes
- mangle symbols (don't keep class/function names)
- don't output comments (well, minus licenses OR bundle them separately, somewhere else than in the source code)
- ditch IE
- Safari 10 also goes out, probably along with 11

Additionally, generate gzip/brotli-encoded chunks:
a) on the run from embedded files (as in a cache warm-up round; smaller binary size [by maybe 2MB? + gz/br encoding algorithms]) and serve from memory
b) during frontend compilation and bundle in the binary along with the non-encoded JS files. You need to respect the `Accept-Encoding` header in that case.

**Comment by stacksmash76** [comment]
The splitting isn't great. Loading small chunks (<10kB) is almost always a waste (additional pointless packet roundabout time) vs. initially loading it with a bigger chunk.

Check out `vite-bundle-visualizer` (or a similar tool) to inspect the bundle. It doesn't make much sense to split on pages which consume a lot of dependency code, which gets bundled to a different chunk.

When you just mindlessly split on pages, you don't get to pick what will get bundled and how. You're pretty much leaving everything to chance, and the chances are, you aren't going to get what you expect or hope for.

node_modules consumes 70% of bundle size. Other 30% come from our code. 99% of the time you can serve gzip. Our code weighs only ~112kB gzipped. Do we split 112kB into chunks? Doesn't make sense.

A far bigger improvement would be a better optimization pass + gzip/brotli guaranteed by our backend (instead of hoping a reverse proxy will handle it for us).

To be more concrete with my comment, React lazy is fine, but don't overdo it. And you should go back to manualChunks since they'll do what you need them to do.

PS. Also, why are you _constantly_ force-pushing?

**Comment by zze0s** [comment]
@fabricionaweb could you update this and resolve the conflicts whenever you get some free time? There's no rush but if it could be done in the upcoming 2-3 weeks that would be good :smile:

**Pull request merged**